apps/flame-defi/app/pool/modules/new-position/components/area-chart-with-range.tsx

```diff
@@ -1,109 +1,101 @@
-          color: "rgba(230, 149, 41, 1)",
-        },
-        data: [
-          120, 132, 101, 134, 90, 230, 210, 180, 222, 150, 210, 300, 250, 400,
-        ],
-      },
-    ],
-    // This dataZoom configuration creates a draggable range‐selector
-    // that covers the entire plotting area. We make it semi‐transparent
-    // so that the chart is still visible behind it.
-    dataZoom: [
-      {
-        type: "slider",
-        show: true,
-        // Fill the full chart area:
-        top: 0,
-        bottom: 0,
-        left: 0,
-        right: 0,
-        height: "100%",
+interface AreaChartWithRangeProps {
+  priceRange?: number[];
+}
 
-        // Default range (from 0% to 100% of x‐axis)
-        start: 0,
-        end: 100,
+export const AreaChartWithRange: React.FC<AreaChartWithRangeProps> = ({
+  // TODO: replace this with the actual price range for the pool
+  priceRange = [1, 30],
+}) => {
+  // Generate sample data for the chart
+  const generateData = () => {
+    const data = [];
+    // Make a more interesting curve with some random variance
+    const baseValues = [
+      20, 18, 22, 20, 23, 25, 21, 19, 22, 24, 22, 26, 27, 25, 28, 30,
+    ];
 
-        // Give it some transparency so the data is visible underneath
-        backgroundColor: "rgba(0, 0, 0, 0.1)",
-        fillerColor: "rgba(0, 0, 0, 0.1)",
-        borderColor: "transparent",
+    for (let i = 0; i < 31; i++) {
+      const index = Math.min(
+        Math.floor((i / 30) * (baseValues.length - 1)),
+        baseValues.length - 2,
+      );
+      const nextIndex = Math.min(index + 1, baseValues.length - 1);
+      const fraction = (i / 30) * (baseValues.length - 1) - index;
 
-        // Turn off extra details and shadows to keep it looking clean
-        showDetail: false,
-        showDataShadow: false,
+      // Interpolate between baseValues with safe access
+      const currentValue = baseValues[index] || 20;
+      const nextValue = baseValues[nextIndex] || 20;
+      const interpolatedValue =
+        currentValue + fraction * (nextValue - currentValue);
 
-        // Make the handles smaller (default is 100%)
-        handleSize: "30%",
+      // Add a small random variance
+      const value = interpolatedValue + (Math.random() * 2 - 1);
 
-        // Hide the top bar
-        moveHandleSize: 0, // Hide the move handle
-        moveHandleIcon: "M 0 0 V 100", // Vertical line from top to bottom
-        moveHandleStyle: {
-          opacity: 0,
-        },
-
-        handleStyle: {
-          color: "rgba(230, 149, 41, 0.5)",
-          borderColor: "rgba(230, 149, 41, 0.5)",
-        },
-        emphasis: {
-          handleStyle: {
-            color: "rgba(230, 149, 41, 1)",
-            borderColor: "rgba(230, 149, 41, 1)",
-          },
-        },
-      },
-    ],
+      data.push({
+        x: i,
+        value,
+      });
+    }
+    return data;
   };
 
+  const data = generateData();
+  const minValue = priceRange?.[0] || 0;
```

Comment:
You also have these default values (0 and 30) in `getPriceRangeArray`. Are they needed here too? Can `AreaChartWithRangeProps.priceRange` actually be required? Or is it from an API?